### ISSUE TITLE
Documentation and typings do not match

### ISSUE THREAD

**Issue by eseliger** [ui-router/core#79]
The documentation says [here](https://ui-router.github.io/ng1/docs/latest/interfaces/ng1.ng1statedeclaration.html#views) in the text that every value in the views object must be a `Ng1ViewDeclaration`. Although, in the example a string is used as well, which works. But the type for views is

```ts
views?: {
    [key: string]: _ViewDeclaration;
}
```
I think the type of values should be  `_ViewDeclaration | string`

Also in `StateDeclaration`, the return type of redirectTo currently is
```ts
RedirectToResult | ((transition: Transition) => Promise<RedirectToResult>);
```
although the documentation says [here](https://ui-router.github.io/ng1/docs/latest/interfaces/ng1.ng1statedeclaration.html#redirectto) it can either be a `RedirectToResult`, or a function returning either a `Promise` or a `RedirectToResult`. So I think it should be 
```ts
RedirectToResult | ((transition: Transition) => RedirectToResult | Promise<RedirectToResult>);
```

**Comment by eseliger** [comment]
Oops, wrong repo, moving this to angular-ui/ui-router

**Comment by christopherthielen** [comment]
Closed by 39044878194fad4ec48e9e9fc166a6feafca14f0
